### PR TITLE
Add files to lesson 2

### PR DESCRIPTION
**Pull request by GrzegorzBajorski**
Signed-off-by: Grzegorz Bajorski <Grzegorz.Bajorski@tieto.com>

### CODE REVIEW

**Comment by GrzegorzBajorski** [comment]
@gradopaw please review

File: students/grzegorz_bajorski/lesson_02_flow_control/adding_factorials.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+n = int(input())
+suma = 0
+fractorial = 1
+
+for i in range(1,n+1):
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/adding_factorials.py]: Missing whitespace after comma

File: students/grzegorz_bajorski/lesson_02_flow_control/factorial.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+n = int(input())
+
+for i in range(1,n):
+    n *= i
+
+print(n)
+
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/factorial.py]: Too many empty lines at the end of the file. According to PEP8 should be only one.

File: students/grzegorz_bajorski/lesson_02_flow_control/chocolate_bar.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+n = int(input())
+m = int(input())
+k = int(input())
+
+print ('YES')  if (k % m == 0 or k % n == 0) and k < m*n  else print('NO')
+
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/chocolate_bar.py]: Too many empty lines at the end of the file. According to PEP8 should be only one.

File: students/grzegorz_bajorski/lesson_02_flow_control/bishop_move.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+x1 = int(input())
+y1 = int(input())
+x2 = int(input())
+y2 = int(input())
+
+print('YES') if (abs(x1-x2) == abs(y1-y2)) else print('NO')
+
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/bishop_move.py]: Too many empty lines at the end of the file. According to PEP8 should be only one.

File: students/grzegorz_bajorski/lesson_02_flow_control/adding_factorials.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    fractorial *= i
+    suma += fractorial
+
+print(suma)
+
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/adding_factorials.py]: Too many empty lines at the end of the file. According to PEP8 should be only one.

File: students/grzegorz_bajorski/lesson_02_flow_control/knight_move.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+x1 = int(input())
+y1 = int(input())
+x2 = int(input())
+y2 = int(input())
+
+z = (x1-x2) * (y1-y2)
+
+print('YES') if (abs(z) == 2) else print('NO')
+
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/knight_move.py]: Too many empty lines at the end of the file. According to PEP8 should be only one.

File: students/grzegorz_bajorski/lesson_02_flow_control/ladder.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+n = int(input())
+
+for j in range(1,n+1):
+    for i in range(1,j+1):
+        print(i, end='')
+    print(sep='\n')
+
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/ladder.py]: Too many empty lines at the end of the file. According to PEP8 should be only one.

File: students/grzegorz_bajorski/lesson_02_flow_control/leap_year.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+print('Enter year number')
+year = int(input())
+
+if (year %4 ==0) and (year % 100 != 0) or (year % 400 == 0):
+	print('LEAF')
+else:
+	print('COMMON')
+
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/leap_year.py]: Too many empty lines at the end of the file. According to PEP8 should be only one.

File: students/grzegorz_bajorski/lesson_02_flow_control/queen_move.py [hunk 9]
```diff
@@ -0,0 +1,7 @@
+x1 = int(input())
+y1 = int(input())
+x2 = int(input())
+y2 = int(input())
+
+print('YES') if (x1-x2 == 0 or y1-y2 == 0 or abs(x1-x2) == abs(y1-y2)) else print('NO')
+
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/queen_move.py]: Too many empty lines at the end of the file. According to PEP8 should be only one.

File: students/grzegorz_bajorski/lesson_02_flow_control/the_number_of_elements_equal_to_the_maximum.py [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+while x != 0:
+    x = int(input())
+    if x > maximum:
+        maximum = x
+        count = 1
+    elif x == maximum:
+        count += 1
+print(count)
+
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/the_number_of_elements_equal_to_the_maximum.py]: Too many empty lines at the end of the file. According to PEP8 should be only one.

File: students/grzegorz_bajorski/lesson_02_flow_control/the_number_of_zero.py [hunk 11]
```diff
@@ -0,0 +1,10 @@
+n = int(input())
+number = 0
+
+for _ in range(n):
+    x = int(input())
+    if x == 0:
+        number += 1
+
+print(number)
+
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/the_number_of_zero.py]: Too many empty lines at the end of the file. According to PEP8 should be only one.

File: students/grzegorz_bajorski/lesson_02_flow_control/the_second_maximum.py [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+while x != 0:
+    x = int(input())
+    if x > maximum:
+        max_second = maximum
+        maximum = x
+    elif x < maximum and x > max_second:
+        max_second = x
+
+print(max_second)
+
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/the_second_maximum.py]: Too many empty lines at the end of the file. According to PEP8 should be only one.

File: students/grzegorz_bajorski/lesson_02_flow_control/chocolate_bar.py [hunk 13]
```diff
@@ -0,0 +1,6 @@
+n = int(input())
+m = int(input())
+k = int(input())
+
+print ('YES')  if (k % m == 0 or k % n == 0) and k < m*n  else print('NO')
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/chocolate_bar.py]: Not needed whitespace after print
Not needed whitespace before if and else

File: students/grzegorz_bajorski/lesson_02_flow_control/leap_year.py [hunk 14]
```diff
@@ -0,0 +1,8 @@
+print('Enter year number')
+year = int(input())
+
+if (year %4 ==0) and (year % 100 != 0) or (year % 400 == 0):
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/leap_year.py]: Not needed whitespaces around operators

File: students/grzegorz_bajorski/lesson_02_flow_control/leap_year.py [hunk 15]
```diff
@@ -0,0 +1,8 @@
...
+	print('LEAF')
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/leap_year.py]: Indentation contains tabs

File: students/grzegorz_bajorski/lesson_02_flow_control/leap_year.py [hunk 16]
```diff
@@ -0,0 +1,8 @@
...
+else:
+	print('COMMON')
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/leap_year.py]: Indentation contains tabs

File: students/grzegorz_bajorski/lesson_02_flow_control/factorial.py [hunk 17]
```diff
@@ -0,0 +1,7 @@
+n = int(input())
+
+for i in range(1,n):
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/factorial.py]: Missing whitespace after comma

File: students/grzegorz_bajorski/lesson_02_flow_control/adding_factorials.py [hunk 18]
```diff
@@ -0,0 +1,10 @@
+n = int(input())
+suma = 0
+fractorial = 1
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/adding_factorials.py]: factorial

File: students/grzegorz_bajorski/lesson_02_flow_control/ladder.py [hunk 19]
```diff
@@ -0,0 +1,7 @@
+n = int(input())
+
+for j in range(1,n+1):
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/ladder.py]: Missing whitespace after comma

File: students/grzegorz_bajorski/lesson_02_flow_control/ladder.py [hunk 20]
```diff
@@ -0,0 +1,7 @@
...
+    for i in range(1,j+1):
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/ladder.py]: Missing whitespace after comma

File: students/grzegorz_bajorski/lesson_02_flow_control/lost_card.py [hunk 21]
```diff
@@ -0,0 +1,11 @@
+n = int(input())
+
+sum_card = 0
+suma = 0
+
+for i in range(1,n):
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/lost_card.py]: Missing whitespace around operator

File: students/grzegorz_bajorski/lesson_02_flow_control/the_second_maximum.py [hunk 22]
```diff
@@ -0,0 +1,14 @@
+maximum = 0
+max_second = 0
+x = 1
+
+while x != 0:
+    x = int(input())
+    if x > maximum:
+        max_second = maximum
+        maximum = x
+    elif x < maximum and x > max_second:
```
Comment (gradopaw) [inline students/grzegorz_bajorski/lesson_02_flow_control/the_second_maximum.py]: could be simplified: max_second < x < maximum

**Pull request merged**